Initial testing framework

mark-burnett opened the pull request:
This is an initial pass at some testing scripts.

mark-burnett committed:
Fix outdated documentation regarding proxies

mark-burnett committed:
Call up.sh as an argument to bash in docs

mark-burnett committed:
Add image build dev script

mark-burnett committed:
Add skeleton for validation scripts

* validate-bootstrap.sh is to be used to validate after genesis
* validate-cluster.sh is to be used to validate a full cluster

sf5715 committed:
This PS adds a promenade yaml that templates the IPs and promenade image.

The promenade yaml will be used by Jenkins and the IPs will be dynamic
so we cannot have them hard-coded.  The version of the Promenade image is also
templated to allow for the Jenkins major.minor.patch tag to replace it.
Creating a generated_configs directory that will be used to store the generated
 yaml and script files.

mark-burnett committed:
Freeze armada version for testing

alanmeadows commented on promenade/templates/generate/validate-cluster.sh:
```diff
@@ -0,0 +1,6 @@
+
+EXPECTED_NODE_COUNT={{ config['Cluster']['nodes'] | length }}
+wait_for_ready_nodes $EXPECTED_NODE_COUNT
+
```
Would we not also validate all pods as part of the promenade deployment are running as well? Nodes could all be ready, but if this is inclusive of skydns, armada, etc. at this stage (as well as any other foundation element) we would want to ensure those are all running and ready in addition to the nodes themselves?

mark-burnett replied:
Hmm,  I was kind of hoping to delay this until we could lean on `armada test`, but we could put something in place for now and replace it with that later.

alanmeadows commented on docs/getting-started.md:
````diff
@@ -74,13 +76,5 @@ docker build --build-arg http_proxy=$http_proxy --build-arg https_proxy=$http_pr
-sudo -E /vagrant/up.sh /vagrant/configs/n0.yaml
-```
+To use Promenade from behind a proxy, set the `http_proxy`, `https_proxy`, and
+`no_proxy` variables in the input configuration fed to the `generate` command.
````
Would it not be helpful on what is likely relevant to be placed into no_proxy, such as the Kubernetes FQDN suffix?

mark-burnett replied:
This had been specified elsewhere in the docs, but it wasn't particularly well-written.

I consolidated those instructions into the configuration doc and refer to them from getting started now.

mark-burnett committed:
Report additional docker state

sf5715 committed:
Update configuration file to replace HostName.

This change will allow for Jenkins to replace the hostname with the newly
provisioned VM's hostname.

mark-burnett committed:
Add missed call to common fail path

mark-burnett committed:
Overwrite /etc/resolv.conf

sh8121att commented on promenade/templates/include/common_validation.sh:
```diff
@@ -0,0 +1,136 @@
+    cat <<EOPOD | kubectl --namespace $NAMESPACE apply -f -
+---
+apiVersion: v1
+kind: Pod
```
Any consideration that this could be done as a DaemonSet so that collecting logs from all nodes is tested?

sh8121att reviewed (approved):
One question/suggestion, otherwise LGTM.
